slang-gfx.h

```diff
@@ -505,6 +505,11 @@ struct ShaderOffset
     SlangInt bindingArrayIndex = 0;
 };
 
+enum class ShaderObjectContainerType
+{
+    None, Array, StructuredBuffer
+};
+
```

Comment:
This has already been discussed separately, but I want to start by noting that a lot of the feedback here doesn't need to be addressed immediately or in the near term. Mostly I'm thinking about where we want the design to land in the long run.

My ideal would be that we don't need this distinction to be part of the API for creating shader objects, for two reasons:

* The user should (eventually) be able to take a reflection type `T` and form the type `T[]` or `StructuredBuffer<T>` using the reflection API, at which point they can directly construct the type they want to form an object for

* Ideally, the user should only need the `Array` case here, and the runtime would allow an object allocated based on an array type to be bound as a structured buffer